Checks/Compose/PatternLengthCheck.cs

```diff
@@ -0,0 +1,219 @@
+                            {
+                                var gapBeginObject = objects.SafeGetIndex(i);
+                                var gapEndObject = objects.SafeGetIndex(i + 1);
+
+                                // Check if gap is smaller than or equal to the snap size
+                                var gap = gapEndObject.time - gapBeginObject.GetEndTime();
+                                if (gap - MS_EPSILON <= snapMs)
+                                {
+                                    foundStartOfPattern = true;
+                                    currentPatternStartTimeMs = gapBeginObject.GetEndTime();
+                                    patternStartIndex = i;
+                                }
+                            }
+
+                            // check if this is the last note of a pattern, and if so check if it's too long
+                            if (foundEndOfPattern && foundStartOfPattern)
+                            {
+                                foundEndOfPattern = false;
+                                foundStartOfPattern = false; // resume checking for start of pattern
+                                var durationOfPattern = i - patternStartIndex + 1;
+
+
+                                if (durationOfPattern > 1 && _DEBUG_SEE_ALL_PATTERN_LENGTHS)
+                                {
+                                    yield return new Issue(
+                                        GetTemplate(_WARNING),
+                                        beatmap,
+                                        Timestamp.Get(currentPatternStartTimeMs).Trim() + ">",
+                                        Timestamp.Get(currentPatternEndTimeMs).Trim() + ">",
+                                        outputDict[snapValues.Key] ?? "unknown snap",
+                                        durationOfPattern
+
+                                    ).ForDifficulties(diff);
+                                }
+                                else if (durationOfPattern > snapValues.Value)
+                                {
+                                    yield return new Issue(
+                                        GetTemplate(_WARNING),
+                                        beatmap,
+                                        Timestamp.Get(currentPatternStartTimeMs).Trim() + ">",
+                                        Timestamp.Get(currentPatternEndTimeMs).Trim() + ">",
+                                        outputDict[snapValues.Key] ?? "unknown snap",
+                                        durationOfPattern
+                                    ).ForDifficulties(diff);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        // Checks if bottom diff is kantan
+        private Boolean isBottomDiffKantan(BeatmapSet beatmapSet)
```

Comment:
better to move this to `TaikoUtils`, makes more sense to have it there + easier to access in case we have to reuse this

Reply:
Moved to `TaikoUtils`